apps/readest-app/src/services/appService.ts

```diff
@@ -291,8 +293,17 @@ export abstract class BaseAppService implements AppService {
         }
       }
     }
-    book.deletedAt = Date.now();
-    book.downloadedAt = null;
+
+    if (
+      (!book.uploadedAt && includingLocal) // file only in local
+      || (!book.downloadedAt && !!book.uploadedAt && includingUploaded) // file only in cloud
+      || (!!book.uploadedAt && includingLocal && includingUploaded) // file both in local and cloud
+    ) {
+      book.deletedAt = Date.now();
```

Comment:
`deletedAt` is only for the local status. We didn't track the deletedAt of the remote file locally. So
```
if (includingLocal) {
      book.deletedAt = Date.now();
      book.downloadedAt = null;
}
```
should be OK.